### PR TITLE
Wagtail 4.2 upgrade

### PR DESCRIPTION
**Pull request by katdom13**
- Include changes from this PR: https://github.com/wagtail/wagtail-autocomplete/pull/130 
- Added Wagtail 4.2 compatibility
- Added tests for Python 3.10 and 3.11
- Remove support for versions of Wagtail < 4.1 (Wagtail 4.1 or later now required)

Note: It is important that this PR be merged first (https://github.com/wagtail/wagtail-autocomplete/pull/130) as it provides support for Wagtail 4.0 while this current PR removes support for versions of Wagtail older than 4.1

All tests run OK

```
collected 24 items                                                                                                                                                                                   

wagtailautocomplete/tests/test_autocomplete_views.py .............                                                                                                                             [ 54%]
wagtailautocomplete/tests/test_edit_handlers.py ...........                                                                                                                                    [100%]
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -53,12 +53,13 @@
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
         'Framework :: Django',
         'Framework :: Django :: 3',
         'Framework :: Django :: 4',
```
Comment (nickmoreton) [inline setup.py:60]:
```suggestion
        'Framework :: Django',
        'Framework :: Django :: 3.2',
        'Framework :: Django :: 4.0',
        'Framework :: Django :: 4.1',
```

File: tox.ini [hunk 2]
```diff
@@ -12,13 +13,15 @@ basepython =
     py37: python3.7
     py38: python3.8
     py39: python3.9
+    py310: python3.10
+    py3.11: python3.11
```
Comment (nickmoreton) [inline tox.ini:17]:
```suggestion
    py311: python3.11
```

File: wagtailautocomplete/tests/test_edit_handlers.py [hunk 3]
```diff
@@ -65,13 +49,12 @@ def test_form_field_media(self):
...
-        result = self.autocomplete_panel.render_as_field()
-
+        result = self.autocomplete_panel.render_html()
         self.assertIn('initAutoCompleteWidget("id_owner");', result)
 
     def test_render_as_field(self):
-        result = self.autocomplete_panel.render_as_field()
-        self.assertIn('<p class="help">the owner</p>', result)
+        result = self.autocomplete_panel.render_html()
+        self.assertIn('<div class="help">the owner</div>', result)
```
Comment (nickmoreton) [inline wagtailautocomplete/tests/test_edit_handlers.py:57]: **Praise**

File: wagtailautocomplete/tests/test_edit_handlers.py [hunk 4]
```diff
@@ -116,27 +93,16 @@ def test_render_as_empty_field(self):
...
-                instance=self.test_house, form=form, request=self.request
-            )
+        autocomplete_panel = edit_handler.children[0].get_bound_panel(
+            instance=self.test_house, form=form, request=self.request
+        )
 
-        result = autocomplete_panel.render_as_field()
-        self.assertIn('Occupants', result)
+        result = autocomplete_panel.render_html()
+        self.assertIn('data-contentpath="occupants"', result)
```
Comment (nickmoreton) [inline wagtailautocomplete/tests/test_edit_handlers.py:105]: Ditto

**Review by nickmoreton — COMMENTED** [review]
Nice work. I left a couple of comments.

File: setup.py (hunk 1 above)
Comment (katdom13) [reply setup.py]: Thanks @nickmoreton,
This should be fixed on this commit https://github.com/wagtail/wagtail-autocomplete/pull/143/commits/b47237304b10f6119d8ef0094f68d08ee02e0163

**Review by harrislapiroff — CHANGES_REQUESTED** [review]
Thanks for the PR @katdom13! Any chance you can get tests working on this branch. Looks like our image is missing newer python versions. It's possible we need a new image for the CircleCI config if that one isn't maintained any longer.

If you get stuck or don't have the bandwidth, we can take a look on our end, just no guarantees about timing.

**Comment by nickmoreton** [comment]
@katdom13 You might want to consider closing this PR. I closed the v4.0 upgrade with a comment of:

> I'm closing this PR as I don't believe it required. The package works OK on wagtail up to v5 when I use it on an upgrade site.

**Comment by katdom13** [comment]
Hi @nickmoreton , closing this. Thanks!

**Pull request closed without merging**